x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/TimestampAware.java

```diff
@@ -0,0 +1,15 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.esql.expression.function;
+
+/**
+ * Marker interface to identify classes of functions that operate on the {code @timestamp} field of an index.
+ * Implementations of this interface need to expect the associated {@code Attribute} to be passed as the following argument after the
+ * {@code Source} one, which is always the first one.
+ */
+public interface TimestampAware {}
```

Comment:
What's the reason for having the Timestamp expression right after Source and not the last Expression, that is:

Class(Source, Expression field, Expression timestamp)
Class(Source, Expression field, Expression timestamp, Configuration config)
Class(Source, Expression field, Expression timestamp, List<Expression>)

In case of ambiguity, this can be sorted out inside the function registry.